maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveJsonSchema.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.maven.derive.schema;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class DeriveJsonSchema extends DeriveSchema {
+
+  public DeriveJsonSchema() {
+    // Map jackson node data type to type understood by json
+    classToDataType.put(com.fasterxml.jackson.databind.node.DoubleNode.class.getName(), "number");
+    classToDataType.put(com.fasterxml.jackson.databind.node.TextNode.class.getName(), "string");
+    classToDataType.put(com.fasterxml.jackson.databind.node.BigIntegerNode.class.getName(),
+        "number");
+    classToDataType.put(com.fasterxml.jackson.databind.node.IntNode.class.getName(), "number");
```

Comment:
nit: ideally we should have these string constants like "number" defined as separate constants in the class and then used here